### PR TITLE
Use Train upper constraints on Python 2

### PR DESCRIPTION
**Pull request by priteau**

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -74,4 +74,8 @@ os_images_upload: True
 os_images_force_rebuild: False
 
 # Whether images should be public (requires admin rights)
-os_images_public: True
\ No newline at end of file
+os_images_public: True
+
+# Use Train upper constraints when running with Python 2, to avoid
+# incompatibility with newer OSC releases.
+os_images_upper_constraints_file: "{% if ansible_python.version.major == '2' %}https://releases.openstack.org/constraints/upper/train{% endif %}"
```
Comment (markgoddard) [inline defaults/main.yml]: It's already on L9

File: meta/main.yml [hunk 2]
```diff
@@ -26,9 +26,9 @@ dependencies:
   - role: stackhpc.os_openstacksdk
     os_openstacksdk_venv: "{{ os_images_venv }}"
     os_openstacksdk_state: "{{ os_images_package_state }}"
-    os_openstacksdk_upper_constraints: "{{ os_images_upper_constraints }}"
+    os_openstacksdk_upper_constraints_file: "{{ os_images_upper_constraints }}"
```
Comment (markgoddard) [inline meta/main.yml]: Should it be os_images_upper_constraints_file?

File: meta/main.yml [hunk 3]
```diff
@@ -26,9 +26,9 @@ dependencies:
...
 
   - role: stackhpc.os-openstackclient
     os_openstackclient_venv: "{{ os_images_venv }}"
     os_openstackclient_state: "{{ os_images_package_state }}"
-    os_openstackclient_upper_constraints: "{{ os_images_upper_constraints }}"
+    os_openstackclient_upper_constraints_file: "{{ os_images_upper_constraints }}"
```
Comment (markgoddard) [inline meta/main.yml]: ditto

**Review by markgoddard — CHANGES_REQUESTED** [review]

File: defaults/main.yml (hunk 1 above)
Comment (priteau) [reply defaults/main.yml]: I didn't notice, thanks!

File: meta/main.yml (hunk 2 above)
Comment (priteau) [reply meta/main.yml]: Good catch.

**Review by markgoddard — APPROVED** [review]

**Pull request merged**